### ISSUE TITLE
False positive for unused/undefined variable in arrow function when used inside double quotes

### ISSUE THREAD

**Issue by ocean90** [sirbrillig/phpcs-variable-analysis#220]
See the following example:

```php
$allowed_extensions = array_map( fn( $extension ) => ".$extension", $allowed_extensions );

// Unused function parameter $extension. (VariableAnalysis.CodeAnalysis.VariableAnalysis.UnusedVariable)
// Variable $extension is undefined. (VariableAnalysis.CodeAnalysis.VariableAnalysis.UndefinedVariable)
```

Fwiw, the same doesn't happen when using a closure.

**Comment by sirbrillig** [comment]
Thank you! It turns out the extra variable detection I had to add for arrow functions was looking at the token's content (`".$extension"` in your example) when it was trying to find the variable name, rather than using the (already extracted and normalized) proper variable name. Fixed in https://github.com/sirbrillig/phpcs-variable-analysis/pull/222
